src/main/java/seedu/address/logic/parser/MakeStatsCommandParser.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TYPE;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.MakeStatsCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new MakeStatsCommand
+ */
+public class MakeStatsCommandParser {
+
+    private static final String TYPE_GENDER = "g";
+    private static final String TYPE_AGE = "a";
+    /**
+     * Parses the given {@code String} of arguments in the context of the MakeStatsCommand
+     * and returns an MakeStatsCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public MakeStatsCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_TYPE);
+
+        Index index;
+
+        //Check if prefix and index are present
+        if (!arePrefixesPresent(argMultimap, PREFIX_TYPE)
```

Comment:
Since we are only looking for a single prefix, maybe we could follow this style (borrowed from Benjamin's sortEvents PR)

```suggestion
        if (!argMultiMap.getValue(PREFIX_TYPE).isPresent()
```

Reply:
ok will change!